neuralpp/symbolic/basic_expression.py

```diff
@@ -215,11 +219,25 @@ def syntactic_eq(self, other) -> bool:
                 return False
 
 
+class BasicAbelianOperation(BasicConstant, AbelianOperation):
```

Comment:
My argument here is that `AbelianOperation` is a subclass of `Constant`: in particular, it is a subclass of "function". Similar to our design that the function `Constant(+)` is a subexpression of the function application `(+ 1 1)`, the `AbelianOperation` is a subexpression of the quantifier expression. And an `AbelianOperation(+)` is a specialize object of function `Constant(+)` in the way that we also know its identity element.

Reply:
This is not very important but just for completeness, here's my counter-argument: just because it is a function, it does not follow that it needs to be wrapper in a Constant. In fact, we have many functions in the system (regular Python functions) that are not wrapped so.

We wrap functions as Constants when we need to create an expression containing them. Now, it is true that in the AbelianOperation, we will often use them in expressions, but not necessarily always. And by making it a Constant, we are tying the class to Expressions, creating an unnecessary dependence. It could be that we might want to use AbelianOperations separately from Expressions someday.

Anyway, given the typical usage we intend, none of this is very important. :-)